plugins/services/src/js/containers/service-connection/ServicePodConnectionEndpointList.js

```diff
@@ -0,0 +1,176 @@
+import React from "react";
+import { routerShape } from "react-router";
+
+import ConfigurationMap from "#SRC/js/components/ConfigurationMap";
+import ConfigurationMapHeading
+  from "#SRC/js/components/ConfigurationMapHeading";
+import ConfigurationMapLabel from "#SRC/js/components/ConfigurationMapLabel";
+import ConfigurationMapRow from "#SRC/js/components/ConfigurationMapRow";
+import ConfigurationMapSection
+  from "#SRC/js/components/ConfigurationMapSection";
+import ConfigurationMapValue from "#SRC/js/components/ConfigurationMapValue";
+import ClipboardTrigger from "#SRC/js/components/ClipboardTrigger";
+import Icon from "#SRC/js/components/Icon";
+
+import ServiceNoEndpointsPanel from "./ServiceNoEndpointsPanel";
+import Service from "../../structs/Service";
+import { getDisplayValue } from "../../utils/ServiceConfigDisplayUtil";
+
+const METHODS_TO_BIND = ["handleOpenEditConfigurationModal", "handleTextCopy"];
+
+class ServicePodConnectionEndpointList extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      copiedCommand: ""
+    };
+
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  handleTextCopy(copiedCommand) {
+    this.setState({ copiedCommand });
+  }
+
+  handleOpenEditConfigurationModal() {
+    const { router } = this.context;
+    router.push(
+      `/services/detail/${encodeURIComponent(this.props.service.getId())}/edit/`
+    );
+  }
+
+  getClipboardTrigger(command) {
+    return (
+      <div className="code-copy-wrapper">
+        <div className="code-copy-icon tight">
+          <ClipboardTrigger
+            className="clickable"
+            copyText={command}
+            onTextCopy={this.handleTextCopy.bind(this, command)}
+            useTooltip={true}
+          >
+            <Icon id="clipboard" size="mini" ref="copyButton" color="grey" />
+          </ClipboardTrigger>
+        </div>
+        {command}
+      </div>
+    );
+  }
+
+  getProtocolValue(portDefinition) {
+    let protocol = portDefinition.protocol || "";
+    if (Array.isArray(protocol)) {
+      protocol = protocol.join(", ");
+    }
+    protocol = protocol.replace(/,\s*/g, ", ");
```

Comment:
I'm pretty sure that protocol is always be an array for pods, thus there is no need to handle the _string_ case here.